ext/hal/st/lib/audio/microphone/OpenPDMFilter.c

```diff
@@ -0,0 +1,314 @@
+/**
+******************************************************************************
+* @file    OpenPDMFilter.c
+* @author  CL
+* @version V1.0.0
+* @date    9-September-2015
+* @brief   Open PDM audio software decoding Library.   
+*          This Library is used to decode and reconstruct the audio signal
+*          produced by ST MEMS microphone (MP45Dxxx, MP34Dxxx). 
+******************************************************************************
+* @attention
+*
+* <h2><center>&copy; COPYRIGHT 2013 STMicroelectronics</center></h2>
+*
+* Licensed under MCD-ST Image SW License Agreement V2, (the "License");
+* You may not use this file except in compliance with the License.
+* You may obtain a copy of the License at:
+*
+*        http://www.st.com/software_license_agreement_image_v2

```

Comment:
This license does not fit.
-1 waiting this issue to be solved.

Reply:
Decision has been taken: we will move to Apache license since:
1. There are no patents associated with the OpenPDM2PCM Library.
    It is just a simple FIR filter.
2. The library code has been written by scratch. So the library code
     HAS NOT been derived from any code external to ST.
3. Black Duck tool didn't find any matching with other external piece of code

Header will be like this:

\/*                                                                                                                                                                                             
 \* Copyright (c) 2018 STMicroelectronics
 \*
 \* SPDX-License-Identifier: Apache-2.0
 \*/